test/hotspot/jtreg/serviceability/jvmti/RedefineClasses/RedefineAnnotations.java

```diff
@@ -86,53 +83,27 @@ public byte[] asm(ClassLoader loader, String className,
                 ProtectionDomain protectionDomain, byte[] classfileBuffer)
             throws IllegalClassFormatException {
 
-            ClassWriter cw = new ClassWriter(0);
-            ClassVisitor cv = new ReAddDummyFieldsClassVisitor(ASM7, cw) { };
-            ClassReader cr = new ClassReader(classfileBuffer);
-            cr.accept(cv, 0);
-            return cw.toByteArray();
-        }
-
-        public class ReAddDummyFieldsClassVisitor extends ClassVisitor {
-
-            LinkedList<F> fields = new LinkedList<>();
-
-            public ReAddDummyFieldsClassVisitor(int api, ClassVisitor cv) {
-                super(api, cv);
-            }
-
-            @Override public FieldVisitor visitField(int access, String name,
-                    String desc, String signature, Object value) {
-                if (name.startsWith("dummy")) {
-                    // Remove dummy field
-                    fields.addLast(new F(access, name, desc, signature, value));
-                    return null;
+            // Shuffle constant pool
+            ClassFile context = ClassFile.of(ClassFile.ConstantPoolSharingOption.NEW_POOL);
+            return context.transformClass(context.parse(classfileBuffer), new ClassTransform() {
+                final List<FieldModel> dummyFields = new ArrayList<>();
+
+                @Override
+                public void accept(ClassBuilder builder, ClassElement element) {
+                    if (element instanceof FieldModel field && field.fieldName().stringValue().startsWith("dummy")) {
+                        // Defer dummy fields to defer their associated constant pool entries
```

Comment:
Nit: This comment is kind of confusing. What does it mean: `defer dummy fields`? Do you mean `defer adding dummy fields to the class file`?